### ISSUE TITLE
[Feature]: Add message to control custom axis

### ISSUE THREAD

**Issue by PBechon** [Simpit-team/KerbalSimpitRevamped#67]
Add a message with to control custom axis

There is 4 custom axis, that could be controlled just like throttle or rotation.

An API seems to exist to update the in [FlightCtrlState ](https://www.kerbalspaceprogram.com/api/class_flight_ctrl_state.html) class.

To be discussed if we do one or 4 messages (with bitmask) to control them.

Feature requested by Catelyn on Discord

**Comment by CdRGit** [comment]
```cs
var vessel = FlightGlobals.ActiveVessel;
for (int i = 0; i < vessel.vesselModules.Count; i++)
{
    if (vessel.vesselModules[i] is AxisGroupsModule)
    {
        var agModule = vessel.vesselModules[i] as AxisGroupsModule;
        agModule.UpdateAxisGroup(KSPAxisGroup.Custom01, (float)myThrottle/ Int16.MaxValue);
    }
}
``` Code snippet to set custom axis group 1 based on throttle value

**Comment by PBechon** [comment]
Another related issue is that the current way to handle the throttle works with rockets, but do not work for some craft where the mainThrottle is assigned to something else (tested with the ArrowE plane where the simpit command does nothing but the shift/ctrl worked).

I could confirm that the code above works to set the throttle for the ArrowE plane but it does not work on regular rockets (tested on Ariane 5).

The following code, taken from kRPC, seems to work with both ArrowE and Ariane 5 : `FlightInputHandler.state.mainThrottle = (float)myThrottle / Int16.MaxValue;`

This code was not tested with custom axis or with other classical axis (yet).

**Comment by CdRGit** [comment]
```cs
public void AutopilotUpdater(FlightCtrlState fcs)
{
    var axisGroupModule = FlightGlobals.ActiveVessel.FindVesselModuleImplementing<AxisGroupsModule>();

    if (myRotation.pitch != 0)
    {
        fcs.pitch = (float)myRotation.pitch / Int16.MaxValue;
        var vessel = FlightGlobals.ActiveVessel;
        axisGroupModule.UpdateAxisGroup(KSPAxisGroup.Pitch, (float)myRotation.pitch / Int16.MaxValue);
    }

    if (myRotation.roll != 0)
    {
        fcs.roll = (float)myRotation.roll / Int16.MaxValue;
        axisGroupModule.UpdateAxisGroup(KSPAxisGroup.Roll, (float)myRotation.roll / Int16.MaxValue);
    }

    if (myRotation.yaw != 0)
    {
        fcs.yaw = (float)myRotation.yaw / Int16.MaxValue;
        axisGroupModule.UpdateAxisGroup(KSPAxisGroup.Yaw, (float)myRotation.yaw / Int16.MaxValue);
    }

    if (myTranslation.X != 0)
    {
        fcs.X = (float)myTranslation.X / Int16.MaxValue;
        axisGroupModule.UpdateAxisGroup(KSPAxisGroup.TranslateX, (float)myTranslation.X / Int16.MaxValue);
    }

    if (myTranslation.Y != 0)
    {
        fcs.Y = (float)myTranslation.Y / Int16.MaxValue;
        axisGroupModule.UpdateAxisGroup(KSPAxisGroup.TranslateY, (float)myTranslation.Y / Int16.MaxValue);
    }

    if (myTranslation.Z != 0)
    {
        fcs.Z = (float)myTranslation.Z / Int16.MaxValue;
        axisGroupModule.UpdateAxisGroup(KSPAxisGroup.TranslateZ, (float)myTranslation.Z / Int16.MaxValue);
    }

    if (myWheel.steer != 0)
    {
        fcs.wheelSteer = (float)myWheel.steer / Int16.MaxValue;
        axisGroupModule.UpdateAxisGroup(KSPAxisGroup.WheelSteer, (float)myWheel.steer / Int16.MaxValue);
    }

    if (myWheel.throttle != 0)
    {
        fcs.wheelThrottle = (float)myWheel.throttle / Int16.MaxValue;
        axisGroupModule.UpdateAxisGroup(KSPAxisGroup.WheelThrottle, (float)myWheel.throttle / Int16.MaxValue);
    }
    
    if (myThrottleFlag)
    {
        FlightInputHandler.state.mainThrottle = (float)myThrottle / Int16.MaxValue;
    }
}
```
this code works for everything, using both fcs and axisGroupModule is not ideal, but does work, I might try to implement custom axes myself as well

**Comment by PBechon** [comment]
@CdRGit Any update on this issue ? Is your code ready to be pulled to the dev branch ? Are you working on the custom axis also ?
I can take over some of it if you need, I just want to avoid duplicating the efforts

**Comment by CdRGit** [comment]
> @CdRGit Any update on this issue ? Is your code ready to be pulled to the dev branch ? Are you working on the custom axis also ? I can take over some of it if you need, I just want to avoid duplicating the efforts

I think it's probably better if you take it over 😅

**Comment by PBechon** [comment]
Custom axis message added with 144f2ff9deb1e6022ecc35efa69df658c03ff470
Fix for already existing axis with 06d83aebef678633e76e147db9f1db68e2b6bcc4
